From f7abbe46870d126a39df8d24c5e56bb7572759e5 Mon Sep 17 00:00:00 2001
From: Anand Mahadevan SS <a0230421@ti.com>
Date: Thu, 19 Sep 2019 11:08:07 +0530
Subject: [PATCH 7/7] arm64: dts: ti: added ION to vision_apps DT overlay

*k3-j721e-vision-apps.dtso
- Replaced CMEM with ION in the DT
- Added dma_buf_phys to the DT
- Renamed carveouts to generic vision_apps instead of ION/CMEM

Signed-off-by: Anand Mahadevan SS <a0230421@ti.com>
---
 arch/arm64/boot/dts/ti/k3-j721e-vision-apps.dtso | 21 +++++++++------------
 1 file changed, 9 insertions(+), 12 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-j721e-vision-apps.dtso b/arch/arm64/boot/dts/ti/k3-j721e-vision-apps.dtso
index 9c899d1..2f2787d 100644
--- a/arch/arm64/boot/dts/ti/k3-j721e-vision-apps.dtso
+++ b/arch/arm64/boot/dts/ti/k3-j721e-vision-apps.dtso
@@ -30,15 +30,15 @@
 				};
 
 			};
+			dma_buf_phys {
+				compatible = "ti,dma_buf_phys";
+			};
 
-			cmem {
-				compatible = "ti,cmem";
-				#address-cells = <1>;
-				#size-cells = <0>;
-				status = "okay";
-				cmem_block_0: cmem_block@0 {
-					reg = <0>;
-					memory-region = <&cmem_shared_region>;
+			ion {
+				compatible = "ti,ion";
+				ti_carveout_heap_0 {
+					memory-region = <&vision_apps_shared_region>;
+					ion-heap-type = <2>;
 				};
 			};
 		};
@@ -63,10 +63,9 @@
 		no-map;
 	};
 
-	cmem_shared_region: cmem-memories@bc000000 {
+	vision_apps_shared_region: vision_apps_shared-memories@bc000000 {
 		compatible = "shared-dma-pool";
 		reg = <0x00 0xbc000000 0x00 0x24000000>;
-		no-map;
 	};
 };
 
@@ -186,5 +185,3 @@
 &main_i2c1 {
 	status = "disabled";
 };
-
-
-- 
1.9.1

